VespaGAS/ViewModels/GasCalculatorViewModel.swift

```diff
@@ -1,23 +1,41 @@
 import Foundation
 import RxSwift
+import ReSwift
 
-struct GasCalculatorViewModel {
-    let oilMixValueObservable: Observable<String>
-    let oilValueObservable: Observable<String>
-    let gasValueObservable: Observable<String>
+class GasCalculatorViewModel: StoreSubscriber {
+    private let oilMixBehaviourSubject = BehaviorSubject(value: OilMix(amount: 0).description)
+    private let oilBehaviourSubject = BehaviorSubject(value: Oil(amount: 0, unit: .Milliliter).description.lowercaseString)
+    private let gasolineBehaviourSubject = BehaviorSubject(value: Gasoline(amount: 0, unit: .Liter).description)
 
-    init(gasObservable: Observable<Gasoline>, oilMixObservable: Observable<OilMix>) {
-        oilMixValueObservable = Observable.combineLatest(gasObservable, oilMixObservable) { (gasoline, oilMix) -> String in
-            let oil = oilMix.oilForGasoline(gasoline)
-            return oil.toUnit(.Milliliter).roundAmount().description.lowercaseString
-        }.startWith(Oil(amount: 0, unit: .Milliliter).description.lowercaseString)
+    var oilMixValueObservable: Observable<String> {
+        get {
+            return oilMixBehaviourSubject.asObservable().distinctUntilChanged()
+        }
+    }
+
+    var  oilValueObservable: Observable<String> {
+        get {
+            return oilBehaviourSubject.asObservable().distinctUntilChanged()
+        }
+    }
+
+    var gasValueObservable: Observable<String> {
+        get {
+            return gasolineBehaviourSubject.asObservable().distinctUntilChanged()
+        }
+    }
+
```

Comment:
In our ViewModel we want to be aware about the changes of the state, so we subscribe to the store.